#! /bin/sh /usr/share/dpatch/dpatch-run
## filecSt9Ia.dpatch by Sebastien Delafond <sdelafond@gmx.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: better handling of the CGI home.

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

diff -urNad zonecheck-2.0.4/installer.rb /tmp/dpep.gXpF1A/zonecheck-2.0.4/installer.rb
--- zonecheck-2.0.4/installer.rb	2005-01-21 16:29:46.000000000 -0800
+++ /tmp/dpep.gXpF1A/zonecheck-2.0.4/installer.rb	2005-01-21 16:29:57.000000000 -0800
@@ -55,7 +55,8 @@
 	ENV['DOCDIR'    ] ||= "#{ENV['PREFIX']}/share/doc"
 	ENV['ETCDIR'    ] ||= "#{ENV['PREFIX']}/etc"
 	ENV['CGIDIR'    ] ||= "#{ENV['LIBEXEC']}/#{ENV['PROGNAME']}/cgi-bin"
-        ENV['WWWDIR'    ] ||= "#{ENV['LIBEXEC']}/#{ENV['PROGNAME']}/www"
+	ENV['WWWDIR'    ] ||= "#{ENV['LIBEXEC']}/#{ENV['PROGNAME']}/www"
+	ENV['WWWCGIDIR' ] ||= "#{ENV['HTML_PATH']}/cgi-bin"
 
 	ENV['VERSION'	] ||= VERSION
 
@@ -165,8 +166,11 @@
     def inst_cli
 	puts "==> Installing CLI"
 	mkdir_p "#{ENV['CHROOT']}#{ENV['BINDIR']}",	:verbose => @verbose
-	ln_s @zc, "#{ENV['CHROOT']}#{ENV['BINDIR']}/#{ENV['PROGNAME']}",
+      @p = pwd
+      cd @ch_installdir, :verbose => @verbose
+      ln_s "../share/zonecheck/zc/zc.rb", "../../bin/#{ENV['PROGNAME']}",
 	    :force => true,				:verbose => @verbose
+      cd @p
 	mkdir_p "#{ENV['CHROOT']}#{ENV['MANDIR']}/man1",:verbose => @verbose
 	install "man/zonecheck.1",
 	    "#{ENV['CHROOT']}#{ENV['MANDIR']}/man1/#{ENV['PROGNAME']}.1",
@@ -177,9 +181,11 @@
 
     def patch_cgi
 	puts "==> Patching HTML pages"
-	Dir["#{@ch_wwwdir}/www/html/*.html.*"].each { |page|
-	    page_content = File.readlines(page)
-	    page_content.each { |line| 
+	Dir["#{@ch_wwwdir}/*.html.*"].each { |page|
+            page_content = File.readlines(page)
+            page_content.each { |line| 
+                line.gsub!(/HTML_PATH\/cgi-bin/, ENV['WWWCGIDIR'])
+                line.gsub!(/\/(fr|en)\/(.*)\"/, "/\\2.\\1\"" )
 		line.gsub!(/HTML_PATH/, ENV['HTML_PATH']) }
 	    File::open(page, "w", 0644) { |io| io.puts page_content }
 	}
@@ -189,15 +195,22 @@
     def inst_cgi
 	puts "==> Installing HTML pages"
 	mkdir_p @ch_wwwdir,				:verbose => @verbose
-	Dir["www/*"].each { |entry|
+	["www/img", "www/js", "www/style"].each { |entry|
 	    cp_r entry, @ch_wwwdir,			:verbose => @verbose
 	}
+	Dir["www/html/*"].each { |entry|
+	    cp entry, @ch_wwwdir,			:verbose => @verbose
+	}
 	puts
 
 	puts "==> Installing CGI"
+      mkdir_p @ch_installdir, :verbose => @verbose
 	mkdir_p "#{ENV['CHROOT']}#{ENV['CGIDIR']}",	:verbose => @verbose
-	ln_s @zc, "#{ENV['CHROOT']}#{ENV['CGIDIR']}/zc.cgi",
+      @p = pwd
+      cd @ch_installdir, :verbose => @verbose
+      ln_s "../../share/zonecheck/zc/zc.rb", "../../lib/cgi-bin/zc.cgi",
 	    :force => true,				:verbose => @verbose
+      cd @p
 	puts
     end
 
